lms/djangoapps/grades/api/views.py

```diff
@@ -128,28 +137,48 @@ def get(self, request, course_id):
         Return:
             A JSON serialized representation of the requesting user's current grade status.
         """
-        username = request.GET.get('username')
-
-        # only the student can access her own grade status info
-        if request.user.username != username:
-            log.info(
-                'User %s tried to access the grade for user %s.',
-                request.user.username,
-                username
-            )
-            return self.make_error_response(
-                status_code=status.HTTP_404_NOT_FOUND,
-                developer_message='The user requested does not match the logged in user.',
-                error_code='user_mismatch'
-            )
 
+        # Use the request user's if none provided.
+
+        if 'username' in request.GET:
+            username = request.GET.get('username')
+        else:
+            username = request.user.username
+
+        # Returns a 404 response if course_id is invalid, or request.user is not enrolled in the course
```

Comment:
Can you say `_get_course() returns a 404 response...` here, assuming that is what the comment is explaining?  Not really clear if the comment applies to the immediately-following code or the overall block which does trigger a 404 on L170